validator/src/commands/run/args/blockstore_options.rs

```diff
@@ -0,0 +1,241 @@
+    };
+
+    #[test]
+    fn verify_args_struct_by_command_run_with_wal_recovery_mode() {
+        // tolerate_corrupted_tail_records
+        {
+            let default_run_args = crate::commands::run::args::RunArgs::default();
+            let expected_args = RunArgs {
+                blockstore_options: BlockstoreOptions {
+                    recovery_mode: Some(BlockstoreRecoveryMode::TolerateCorruptedTailRecords),
+                    ..default_run_args.blockstore_options.clone()
+                },
+                ..default_run_args.clone()
+            };
+            verify_args_struct_by_command_run_with_identity_setup(
+                default_run_args,
+                vec!["--wal-recovery-mode", "tolerate_corrupted_tail_records"],
+                expected_args,
+            );
+        }
+
+        // absolute_consistency
+        {
+            let default_run_args = crate::commands::run::args::RunArgs::default();
+            let expected_args = RunArgs {
+                blockstore_options: BlockstoreOptions {
+                    recovery_mode: Some(BlockstoreRecoveryMode::AbsoluteConsistency),
+                    ..default_run_args.blockstore_options.clone()
+                },
+                ..default_run_args.clone()
+            };
+            verify_args_struct_by_command_run_with_identity_setup(
+                default_run_args,
+                vec!["--wal-recovery-mode", "absolute_consistency"],
+                expected_args,
+            );
+        }
+
+        // point_in_time
+        {
+            let default_run_args = crate::commands::run::args::RunArgs::default();
+            let expected_args = RunArgs {
+                blockstore_options: BlockstoreOptions {
+                    recovery_mode: Some(BlockstoreRecoveryMode::PointInTime),
+                    ..default_run_args.blockstore_options.clone()
+                },
+                ..default_run_args.clone()
+            };
+            verify_args_struct_by_command_run_with_identity_setup(
+                default_run_args,
+                vec!["--wal-recovery-mode", "point_in_time"],
+                expected_args,
+            );
+        }
+
+        // skip_any_corrupted_record
+        {
+            let default_run_args = crate::commands::run::args::RunArgs::default();
+            let expected_args = RunArgs {
+                blockstore_options: BlockstoreOptions {
+                    recovery_mode: Some(BlockstoreRecoveryMode::SkipAnyCorruptedRecord),
+                    ..default_run_args.blockstore_options.clone()
+                },
+                ..default_run_args.clone()
+            };
+            verify_args_struct_by_command_run_with_identity_setup(
+                default_run_args,
+                vec!["--wal-recovery-mode", "skip_any_corrupted_record"],
+                expected_args,
+            );
+        }
+    }
+
+    #[test]
+    fn verify_args_struct_by_command_run_with_rocksdb_ledger_compression() {
```

Comment:
Since we have manual parsing, maybe should have a test for invalid value too ?

Reply:
sure! [f2ab894](https://github.com/anza-xyz/agave/pull/6918/commits/f2ab8941a4bdae51095bbb8c1e5981fad2963384)